docs/content/release-notes/flink-2.0.md

```diff
@@ -0,0 +1,1666 @@
+
+See DataStream API(V2) [Documentation](https://nightlies.apache.org/flink/flink-docs-release-2.0/docs/dev/datastream-v2/overview/) for more details
+
+**NOTICE:** The new DataStream API is currently in the experimental stage and is not yet stable, thus not recommended for production usage at the moment.
+
+#### Java Supports
+
+##### [FLINK-33163](https://issues.apache.org/jira/browse/FLINK-33163)
+
+Starting the 2.0 version, Flink officially supports Java 21.
+
+The default and recommended Java version is changed to Java 17 (previously Java 11). This change mainly affect the docker images and building Flink from sources.
+
+Meanwhile, Java 8 is no longer supported.
+
+#### Serialization Improvements
+
+##### [FLINK-34037](https://issues.apache.org/jira/browse/FLINK-34037)
+
+Flink 2.0 introduces much more efficient built-in serializers for collection types (i.e., Map / List / Set), which is enabled by default.
+
+We have also upgraded Kryo to version 5.6, which is faster and more memory efficient, and has better supports for newer Java versions.
+
+# Highlight Breaking Changes
+
+## API
+
+The following sets of APIs have been completely removed.
+- **DataSet API** - Please migrate to DataStream API, or Table API/SQL if applicable. See also "How to Migrate from DataSet to DataStream".
+- **Scala DataStream and DataSet API** - Please migrate to the Java DataStream API.
+- **SourceFuction, SinkFunction and Sink V1** - Please migrate to Source and Sink V2.
+- **TableSoure and TableSink** - Please migrate to DynamicTableSource and DynamicTableSink. See also "User-defined Sources & Sinks".
+- **TableSchema, TableColumn and Types** - Please migrate to Schema, Column and DataTypes respectively.
+
+Some deprecated methods have been removed from DataStream API. See also the list of breaking programming APIs.
+
+Some deprecated fields have been removed from REST API. See also the list of breaking REST APIs.
+
+**NOTICE:** You may find some of the removed APIs still exist in the code base, usually in a different package. They are for internal usages only and can be changed / removed anytime without notifications. Please **DO NOT USE** them.
+
+### Connector Adaption Plan
+
+As SourceFunction, SinkFunction and SinkV1 being removed, existing connectors depending on these APIs will not work on the Flink 2.x series. Here’s the plan for adapting the first-party connectors.
+- A new version of Kafka, Paimon, JDBC and ElasticSearch connectors, adapted to the API changes, will be released right after the release of Flink 2.0.0.
+- We plan to gradually migrate the remaining first-party connectors within 3 subsequent minor releases (i.e., by Flink 2.3).
```

Comment:
Do we have a link to first-party connectors - I have not heard that term before.

Reply:
that means connectors officially maintained by community.

Reply:
I wonder if we could use that phrase (connectors officially maintained by community ) maybe with a link to our connector docs. As first-party connectors could be ambiguous as it is not defined.